### PR TITLE
fix: Analyze the output of StarCoder-1B for python

### PR DESCRIPTION
**Pull request by yan91083**

### CODE REVIEW

**Comment by wsxiaoys** [comment]
Thanks for the pr! The metric computation looks great! Could you also checkin the code computing output_starcoder_1B_rg1_bm25.jsonl?

**Comment by yan91083** [comment]
I used the same tabby.py and replaced the sample.jsonl by line_completion_rg1_bm25.jsonl

**Comment by wsxiaoys** [comment]
> I used the same tabby.py and replaced the sample.jsonl by line_completion_rg1_bm25.jsonl

Great, let's proceed to following tasks:
1. creating dagster orchastration to make it easy to run evaluation on one languages on all settings (line_completion, line_completion_oracle, line_completion_bm25)
2. creating jupyter visualization of output for ease for quality investigation

File: experimental/eval/analyze.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    output_file = "./data/" + model + "/" + language + "/result_" + file[7:]
+
+    with open(output_file, "w") as fout:
+        with open(input_file, "r") as fin:
+            for line in fin:
+                obj = json.loads(line)
+                prompt = obj["prompt"]
+                groundtruth = obj["label"]
+                prediction = obj["prediction"]
+                first_line_groundtruth = groundtruth.split("\n")[0].strip()
```
Comment (wsxiaoys) [inline experimental/eval/analyze.py:16]: we shouldn't do strip here...

File: experimental/eval/analyze.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                first_line_prediction = prediction.split("\n")[0].strip()
```
Comment (wsxiaoys) [inline experimental/eval/analyze.py:17]: ditto

Comment (wsxiaoys) [inline experimental/eval/analyze.py:17]:
```suggestion
                first_line_groundtruth = groundtruth.split("\n")[0]
                first_line_prediction = prediction.split("\n")[0]
```

File: experimental/eval/predict.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+    objs = []
+    with open(whole_path_file) as fin:
+        for line in fin:
+            obj = json.loads(line)
+            objs.append(obj)
+
+    df = pd.DataFrame(objs)
+
+    outputs = await asyncio.gather(*[model.complete.remote.aio(language, crossfile_context, index, row) for index, row in df.iterrows()])
```
Comment (wsxiaoys) [inline experimental/eval/predict.py:163]: this is fine, but you might wanna look at `return_exceptions` in https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

Which could probably further simplify your implementation

**Comment by wsxiaoys** [comment]
Please split the python code into separate PR (this pr is too huge to view: nothing displayed at https://github.com/TabbyML/tabby/pull/642/files)

**Comment by yan91083** [comment]
I reset the branch and reverted several commits. I think now you can see the file changes.

**Pull request closed without merging**